core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -61,6 +62,12 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
   // and AccumProfileResults)
   val stageToDiagnosticMetrics: HashMap[Long, HashMap[String, AccumProfileResults]] =
     HashMap.empty[Long, HashMap[String, AccumProfileResults]]
+  // A map between (sql ID, node ID, node name, stage IDs) and a list of IO diagnostic metrics
+  // results (stored as an array of SQLAccumProfileResults) - we need all info in the key to
+  // uniquely identify each SQLAccumProfileResults
+  val IODiagnosticMetrics:
```

Comment:
- Can we include `Map` as suffix in the variable name? It would help in conveying that it is a map.
- If the string `stageIds` become very lengthy, it might make the key very long (and increase memory usage). Should we use a hash value to represent stageIds instead of storing the entire string?

Reply:
> Can we include Map as suffix in the variable name? It would help in conveying that it is a map.
thanks @parthosa , updated this

> If the string stageIds become very lengthy, it might make the key very long (and increase memory usage). Should we use a hash value to represent stageIds instead of storing the entire string?
The stageIds should not be lengthy, I believe there will be at most 2 stages for each SQL node. @nartal1 might have more knowledge about this.